### PR TITLE
Fix redirect url redirect_url  when OIDC auth mode is enabled

### LINKED ISSUE #17594: redirect_url is not working when OIDC is enabled
Harbor does has support for `redirect_url` to redirect the user to the page he planned to go before hitting the login page. 

However `redirect_url` is not working and the value is ignored.

- Redirect is not set
- If redirect is set it is ignored

### PR DESCRIPTION
**Pull request by tpoxa**
Fix redirect URL parameter `redirect_url` in case the user is redirected to sign-in screen in OIDC.

The fix is using server-side session storage to track the original URL being requested before login redirect.

# Issue being fixed
Fixes #17594

Please indicate you've done the following:
- [x] Well Written Title and Summary of the PR
- [x] Label the PR as needed. **release-note/enhancement**
- [x] Accepted the DCO. Commits without the DCO will delay acceptance.
- [x] Made sure tests are passing and test coverage is added if needed.
- [x] Considered the docs impact and opened a new docs issue or PR with docs changes if needed in [website repository](https://github.com/goharbor/website).

### CODE REVIEW

**Review by Vad1mo — APPROVED** [review]

**Comment by AllForNothing** [comment]
@tpoxa Please go into the `portal` directory, and run `npm run lint_fix` to lint your ts code, this will help to pass the ci

File: src/core/controllers/oidc.go [hunk 1]
```diff
@@ -35,6 +35,7 @@ import (
 const tokenKey = "oidc_token"
 const stateKey = "oidc_state"
 const userInfoKey = "oidc_user_info"
+const redirectUrlKey = "oidc_redirect_url"
```
Comment (stonezdj) [inline src/core/controllers/oidc.go:38]: const redirectUrlKey should be redirectURLKey

File: src/core/controllers/oidc.go [hunk 2]
```diff
@@ -85,7 +87,12 @@ func (oc *OIDCController) Callback() {
 		oc.SendBadRequestError(errors.Errorf("OIDC callback returned error: %s - %s", errorCode, errorDescription))
 		return
 	}
-
+	var redirectUrlStr string
```
Comment (stonezdj) [inline src/core/controllers/oidc.go:90]: var redirectUrlStr should be redirectURLStr (revive)

File: src/core/controllers/oidc.go [hunk 3]
```diff
@@ -85,7 +87,12 @@ func (oc *OIDCController) Callback() {
...
+	redirectUrl := oc.GetSession(redirectUrlKey)
```
Comment (stonezdj) [inline src/core/controllers/oidc.go:91]: var redirectUrl should be redirectURL (revive)
see: https://github.com/goharbor/harbor/actions/runs/3255311148/jobs/5344498524

**Review by AllForNothing — APPROVED** [review]
LGTM

**Pull request merged**